src/app/games/trexo/TrexoTutorial.ts

```diff
@@ -0,0 +1,74 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPValidation } from 'src/app/utils/MGPValidation';
+import { TrexoMove } from './TrexoMove';
+import { TrexoRules } from './TrexoRules';
+import { TrexoState } from './TrexoState';
+
+const initialState: TrexoState = TrexoState.getInitialState();
+const moves: TrexoMove[] = [
+    TrexoMove.from(new Coord(4, 4), new Coord(4, 3)).get(),
+    TrexoMove.from(new Coord(6, 4), new Coord(5, 4)).get(),
+    TrexoMove.from(new Coord(5, 5), new Coord(5, 6)).get(),
+    TrexoMove.from(new Coord(4, 4), new Coord(5, 4)).get(),
+    TrexoMove.from(new Coord(7, 3), new Coord(8, 3)).get(),
+    TrexoMove.from(new Coord(3, 1), new Coord(3, 2)).get(),
+    TrexoMove.from(new Coord(7, 5), new Coord(6, 5)).get(),
+];
+let previousState: TrexoState = initialState;
+const movesAndStates: { move: TrexoMove, state: TrexoState }[] = [];
+for (const move of moves) {
+    const state: TrexoState = TrexoRules.get().applyLegalMove(move, previousState);
+    movesAndStates.push({ move, state });
+    previousState = state;
+}
+const firstMove: TrexoMove = TrexoMove.from(new Coord(5, 5), new Coord(5, 4)).get();
+const stateAtTurnOne: TrexoState = TrexoRules.get().applyLegalMove(firstMove, initialState);
+const secondMove: TrexoMove = TrexoMove.from(new Coord(5, 7), new Coord(5, 6)).get();
+const stateAtTurnTwo: TrexoState = TrexoRules.get().applyLegalMove(secondMove, stateAtTurnOne);
+
+export class TrexoTutorial extends Tutorial {
+    public tutorial: TutorialStep[] = [
+        // 1. but du jeu et plateau
+        TutorialStep.informational(
+            $localize`Goal of the game`,
+            $localize`At Trexo, the goal of the game is to align 5 piece of your color in an horizontal, vertical or diagonal line. But player must put a tile on the board, and that tile is constituted of two pieces, one from each players!`,
+            TrexoState.getInitialState()),
+        TutorialStep.anyMove(
+            $localize`Dropping a tile`,
+            $localize`When you drop at tile, it need to be on even ground, and it cannot be right on top of another tile. In others words, it need to be either on the floor, either on two tiles at the same height. To do that, just click on the place where you want to put the opponent piece, then on the neighboring square where you want to put your piece.<br/><br/>You're playing Dark, go ahead.`,
+            TrexoState.getInitialState(),
+            TrexoMove.from(new Coord(4, 4), new Coord(3, 4)).get(),
+            $localize`Congratulations!`),
+        TutorialStep.fromMove(
+            $localize`Put a piece over other pieces`,
+            $localize`When two different pieces on the same level touch each other, you can put a piece on them. You must respect two rules for that:<br/><ul><li>The two pieces must be at the same level.</li><li>You cannot drop your piece on only one other piece.</li></ul><br/>You're playing Dark, do such a move!`,
```

Comment:
hmm c'est litérallement : "quand condition 1 et condition 2, vous pouvez mettre une pièce. Vous devez respectez deux conditions : condition 1 et condition 2", non ?
Tu pourrais directement faire : "You can drop a piece on top of other pieces, if you respect the following conditions: a and b"

Reply:
Oui haha, formulation nulle !